openmmtools/multistate/multistatesampler.py

```diff
@@ -1373,31 +1370,15 @@ def _minimize_replica(self, replica_id, tolerance, max_iterations):
             replica_id + 1, self.n_replicas, initial_energy))
 
         # Minimize energy.
-        try:
-            if max_iterations == 0:
-                logger.debug('Using FIRE: tolerance {} minimizing to convergence'.format(tolerance))
-                while integrator.getGlobalVariableByName('converged') < 1:
-                    integrator.step(50)
-            else:
-                logger.debug('Using FIRE: tolerance {} max_iterations {}'.format(tolerance, max_iterations))
-                integrator.step(max_iterations)
-        except Exception as e:
-            if str(e) == 'Particle coordinate is nan':
-                logger.debug('NaN encountered in FIRE minimizer; falling back to L-BFGS after resetting positions')
-                sampler_state.apply_to_context(context)
-                openmm.LocalEnergyMinimizer.minimize(context, tolerance, max_iterations)
-            else:
-                raise e
+        openmm.LocalEnergyMinimizer.minimize(context, tolerance, max_iterations)
 
         # Get the minimized positions.
         sampler_state.update_from_context(context)
 
         # Compute the final energy of the system for logging.
         final_energy = thermodynamic_state.reduced_potential(sampler_state)
-        logger.debug('Replica {}/{}: final energy {:8.3f}kT'.format(
-            replica_id + 1, self.n_replicas, final_energy))
+        logger.debug(f'Replica {replica_id + 1}/{self.n_replicas}: final energy {final_energy:8.3f}kT')
         # TODO if energy > 0, use slower openmm minimizer
-
         # Clean up the integrator
         del context
```

Comment:
@ijpulidos : I think we might need to remove the `del context` if we're using `self.energy_context_cache.get_context()` since this would delete a `Context` object under the management of the context cache. I can't recall what the correct behavior is here---I think it was to leave the `Context` under management.

This was perhaps different for the FIRE minimizer since a different integrator would lead to a different `Context` we would not re-use.

Reply:
That is a good point, I think we should be okay by removing the `del context` here.